opentelemetry-api/src/opentelemetry/context/propagation/httptextformat.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+from opentelemetry.trace import SpanContext
+
+Setter = typing.Callable[[object, str, str], None]
+Getter = typing.Callable[[object, str], str]
+
+
+class HTTPTextFormat(abc.ABC):
+    """API for propagation of spans via headers.
+
+    This class provides an interface that enables extracting and injecting
+    trace state into headers of HTTP requests. Http frameworks and client
```

Comment:
I suggest that we pick another name than "trace state" to avoid the confusion with the W3C [tracestate](https://www.w3.org/TR/trace-context/#tracestate-field).

Reply:
right! will just call it span context.